### DISCUSSION TITLE
Is there a way to use a custom map in a geoplot?

### DISCUSSION THREAD

**Discussion by lubeskih** [alandefreitas/matplotplusplus#145] [Q&A] [answered]
I would like to use my own vector map of the Earth in a geoplot, is this possible, and if so, how?

**Comment by alandefreitas** [comment, marked as answer]
The geographic functions [usually](https://github.com/alandefreitas/matplotplusplus/blob/ff5c42a6d53ead36a2efe5be2f61903cbcb8bd52/source/matplot/core/axes_type.cpp#L3822) call [`geoplot()`](https://github.com/alandefreitas/matplotplusplus/blob/ff5c42a6d53ead36a2efe5be2f61903cbcb8bd52/source/matplot/core/axes_type.cpp#L3788), which [ensures the figure has a map](https://github.com/alandefreitas/matplotplusplus/blob/ff5c42a6d53ead36a2efe5be2f61903cbcb8bd52/source/matplot/core/axes_type.cpp#L3791) in it, or [creates a map](https://github.com/alandefreitas/matplotplusplus/blob/ff5c42a6d53ead36a2efe5be2f61903cbcb8bd52/source/matplot/core/axes_type.cpp#L3797) otherwise by calling the [`world_map_110m()`](https://github.com/alandefreitas/matplotplusplus/blob/ff5c42a6d53ead36a2efe5be2f61903cbcb8bd52/source/matplot/core/axes_type.cpp#L3797) which returns the map points. Everything else is just calling the line plot functions to draw the map.

So if you have your own map, all you need to do is bypass these points and call plot with your own points instead. Something like:

```cpp
        auto [x, y] = your_own_vector_map();
        line_handle a = ax->plot(x, y);
        a->tag("map");
        color_array land_color = {0, 0.9294, 0.9294, 0.9294};
        a->color(land_color);
        a->fill(true);
        color_array bg = {0, 0.7882, 0.7882, 0.7882};
        ax->color(bg);

        ax->x_axis().geographic(true);
        ax->x_axis().tick_label_format("%Dº%E");
        ax->x_axis().label("Longitude");
        ax->y_axis().geographic(true);
        ax->y_axis().tick_label_format("%Dº%N");
        ax->y_axis().label("Latitude");
```

The [geolimits](https://github.com/alandefreitas/matplotplusplus/blob/ff5c42a6d53ead36a2efe5be2f61903cbcb8bd52/source/matplot/core/axes_type.cpp#L3832) function will later crop unused data points from the map.

**Reply by lubeskih** [reply]
Understood. Thank you very much! :)
